src/main/java/stsjorbsmod/memories/HumilityMemory.java

```diff
@@ -56,7 +56,7 @@ private void updateAppliedThorns() {
             thornsAlreadyApplied = 0;
         }
         
-        int goalBonusThorns = existingThorns - thornsAlreadyApplied;
+        int goalBonusThorns = 2;
```

Comment:
I think it would be better to simplify this implementation now that it doesn't have to maintain the dynamic calculation; `onGainPassiveEffect` and `onLosePassiveEffect` can just invoke an apply/reduce power action for `THORNS_PASSIVE` thorns and drop all of the fancier calculate/update thorns logic.